pkg/operator/ceph/object/spec.go

```diff
@@ -113,8 +113,9 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 			controller.DaemonVolumes(c.DataPathMap, rgwConfig.ResourceName),
 			c.mimeTypesVolume(),
 		),
-		HostNetwork:       c.clusterSpec.Network.IsHost(),
-		PriorityClassName: c.store.Spec.Gateway.PriorityClassName,
+		ServiceAccountName: "rook-ceph-rgw",
```

Comment:
If we are creating a separate service account for RGW, I would first make a PR to do that separately from this.